sdk/protocol-manager-poc/src/implementation/aaveV3Plugin.ts

```diff
@@ -0,0 +1,167 @@
+import { AddressValue, Percentage, TokenAmount, TokenSymbol, Price, CurrencySymbol, RiskRatio } from "@summerfi/sdk-common/common"
+import type { AaveV3LendingPool, AaveV3PoolDebtConfig, AaveV3PoolCollateralConfig, AaveV3PoolId } from "@summerfi/sdk-common/protocols"
+import { PoolType, ProtocolName, EmodeType } from "@summerfi/sdk-common/protocols"
+import { Position, Token } from "@summerfi/sdk-common/common"
+import { BigNumber } from 'bignumber.js'
+import { z } from 'zod'
+import { ChainId, CreateProtocolPlugin, IPositionId, ProtocolManagerContext, ProtocolPlugin } from "../interfaces";
+import { AaveV3LikePluginBuilder, filterAssetsListByEMode } from "./aaveV3LikeBuilder";
+import { UNCAPPED_SUPPLY, PRESISION_BI } from "./constants";
+
+
+function tokenAmountFromBaseUnit({amount, token}: {amount: string, token: Token}): TokenAmount {
+    return TokenAmount.createFromBaseUnit({token, amount})
+}
+
+export const createAaveV3Plugin: CreateProtocolPlugin<AaveV3PoolId> = (ctx: ProtocolManagerContext): ProtocolPlugin<AaveV3PoolId> => {
+    const plugin: ProtocolPlugin<AaveV3PoolId> = {
+        supportedChains: [ChainId.Mainnet],
+        getPool: async (aaveV3PoolId: unknown): Promise<AaveV3LendingPool> => {
+            plugin._validate(aaveV3PoolId)
+            const emode = aaveV3EmodeCategoryMap[aaveV3PoolId.emodeType]
+            if (!emode && emode !== 0n) throw new Error('emode on poolId not recognised')
+
+            const chainId = ctx.provider.chain?.id
+            if (!chainId) throw new Error('ctx.provider.chain.id undefined')
+
+            if (!plugin.supportedChains.includes(chainId)) {
+                throw new Error(`Chain ID ${chainId} is not supported`);
+            }
+
+            const builder = await (new AaveV3LikePluginBuilder(ctx, aaveV3PoolId.protocol.name)).init();
+            const reservesAssetsList = await builder
```

Comment:
I like the chaining. If my other comments about refactoring the return types is getting in the way of this, please ignore them. Perhaps a README with a proper explanation on what is the idea behind it is better

Reply:
No problem. I was planning on doing that as part of the integration work into protocol-plugins directory if that works for you

Reply:
Sure! It works